framework/src/transfers/MultiAppMeshFunctionTransfer.C

```diff
@@ -219,7 +219,8 @@ MultiAppMeshFunctionTransfer::execute()
   {
     if (i_proc == processor_id())
       continue;
-    _communicator.send(i_proc, outgoing_points[i_proc]);
+    std::vector<Parallel::Request> reqs(n_processors());
```

Comment:
I don't understand why we ever need to call `wait` on them.  The only thing that would do is move an error from processor A when it calls `communicator.receive` to processor B when it calls `request.wait`, right?